packages/transport/src/api/usb.ts

```diff
@@ -395,10 +393,28 @@ export class UsbApi extends AbstractApi {
         return device.device;
     }
 
-    private createDevices(nonHidDevices: USBDevice[], signal?: AbortSignal) {
+    private async createDevices(
+        nonHidDevices: USBDevice[],
```

Comment:
we can get rid of code duplicates from `enumetate` and `listen` functions by passing all the devices here and do filterDevices before const loadedDevices = await Promise.all(

Reply:
cool, added as a new commit 5d893eedb877d12d5e9ec17a85ad5c5aa7885a7c